pkg/server/registry/apigroups/acorn/projects/strategy.go

```diff
@@ -66,7 +70,18 @@ func (s *Strategy) Create(ctx context.Context, object types.Object) (types.Objec
 }
 
 func (s *Strategy) Update(ctx context.Context, object types.Object) (types.Object, error) {
-	return s.updater.Update(ctx, object)
+	project := object.(*apiv1.Project)
+	duplicateRegion := make(map[string]bool)
+	for i := 0; i < len(project.Spec.SupportedRegions); i++ {
+		if duplicateRegion[project.Spec.SupportedRegions[i]] {
+			project.Spec.SupportedRegions = append(project.Spec.SupportedRegions[:i], project.Spec.SupportedRegions[i+1:]...)
+			i--
+		} else {
+			duplicateRegion[project.Spec.SupportedRegions[i]] = true
+		}
+	}
+	sort.Strings(project.Spec.SupportedRegions)
```

Comment:
Let's not do this. If a user adds a duplicate region, then that is on them.

Secondly, it's generally bad for us to modify things on `Spec` from the api-server or controller.

Reply:
spec is more concrete vs status?

Reply:
Spec is defined by the user. If we change the spec before saving it to Kubernetes, then GitOps tools like flux will always think that the object is out of date and needs to be updated.